### PR TITLE
feat: support use default browser open source content

### PR DESCRIPTION
**Pull request by BlackHole1**
### Description

Support adding an option in the settings page to allow users to choose whether to open the original content through the default browser.

### Additional context

Close: https://discord.com/channels/1243823539426033696/1294094455485366312

### CODE REVIEW

File: apps/renderer/src/atoms/settings/general.ts [hunk 1]
```diff
@@ -42,6 +43,7 @@ export const subscribeShouldUseIndexedDB = (callback: (value: boolean) => void)
 
 export const generalServerSyncWhiteListKeys: (keyof GeneralSettings)[] = [
   "appLaunchOnStartup",
+  "useDefaultBrowserOpenSourceContent",
```
Comment (Innei) [inline apps/renderer/src/atoms/settings/general.ts:46]: Remove this, This item can be sync to server

File: apps/renderer/src/atoms/settings/general.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { createSettingAtom } from "./helper"
 const createDefaultSettings = (): GeneralSettings => ({
   // App
   appLaunchOnStartup: false,
+  useDefaultBrowserOpenSourceContent: false,
```
Comment (Innei) [inline apps/renderer/src/atoms/settings/general.ts:10]: Move to bottom, This item should not be categorized as an App

File: apps/renderer/src/hooks/biz/useEntryActions.tsx [hunk 3]
```diff
@@ -166,6 +167,9 @@ export const useEntryActions = ({
   const enableInstapaper = useIntegrationSettingKey("enableInstapaper")
   const instapaperUsername = useIntegrationSettingKey("instapaperUsername")
   const instapaperPassword = useIntegrationSettingKey("instapaperPassword")
+  const useDefaultBrowserOpenSourceContent = useGeneralSettingKey(
```
Comment (Innei) [inline apps/renderer/src/hooks/biz/useEntryActions.tsx:170]: There is no need to observe to this setting item.

Comment (Innei) [reply apps/renderer/src/hooks/biz/useEntryActions.tsx:170]: Remove this code

File: apps/renderer/src/hooks/biz/useEntryActions.tsx [hunk 4]
```diff
@@ -396,6 +400,12 @@ export const useEntryActions = ({
         active: showSourceContent,
         onClick: () => {
           if (!populatedEntry.entries.url) return
+
+          if (useDefaultBrowserOpenSourceContent) {
```
Comment (Innei) [inline apps/renderer/src/hooks/biz/useEntryActions.tsx:404]:
```suggestion
          if (getGeneralSettings().useDefaultBrowserOpenSourceContent) {
```

File: apps/renderer/src/hooks/biz/useEntryActions.tsx [hunk 5]
```diff
@@ -480,6 +490,7 @@ export const useEntryActions = ({
     showSourceContentModal,
     read,
     unread,
+    useDefaultBrowserOpenSourceContent,
```
Comment (Innei) [inline apps/renderer/src/hooks/biz/useEntryActions.tsx:493]: Remove this code

File: apps/renderer/src/modules/settings/tabs/general.tsx [hunk 6]
```diff
@@ -53,6 +53,10 @@ export const SettingGeneral = () => {
     setGeneralSetting("appLaunchOnStartup", checked)
   }, [])
 
+  const saveSourceContentSetting = useCallback((checked: boolean) => {
```
Comment (Innei) [inline apps/renderer/src/modules/settings/tabs/general.tsx:56]: This `useCallback` is unnecessary, remove this wrap or just write function inline

**Review by Innei — CHANGES_REQUESTED** [review]

File: apps/renderer/src/atoms/settings/general.ts (hunk 2 above)
Comment (Innei) [reply apps/renderer/src/atoms/settings/general.ts:10]: And this name is too lang

Comment (BlackHole1) [reply apps/renderer/src/atoms/settings/general.ts:10]: > Move to bottom, This item should not be categorized as an App

Which category do you recommend placing it in? Perhaps we should add a new category? Such as _interaction_ or _misc_.

> And this name is too long.

What do you think of `openContentInBrowser`?

File: apps/renderer/src/modules/settings/tabs/general.tsx [hunk 7]
```diff
@@ -72,6 +76,13 @@ export const SettingGeneral = () => {
             },
           }),
           LanguageSelector,
+          defineSettingItem("useDefaultBrowserOpenSourceContent", {
+            label: t("general.use_default_browser_open_source_content"),
+            disabled: !tipcClient,
```
Comment (Innei) [inline apps/renderer/src/modules/settings/tabs/general.tsx:81]: Remove this

File: apps/renderer/src/hooks/biz/useEntryActions.tsx (hunk 4 above)
Comment (Innei) [reply apps/renderer/src/hooks/biz/useEntryActions.tsx:404]: The logic here is incorrect. You need to make a choice with `openInBrowser` above

**Review by Innei — CHANGES_REQUESTED** [review]

**Comment by BlackHole1** [comment]
> You need to make a choice with openInBrowser above

I don't quite understand the meaning of this sentence.

BTW. I think this feature might not be suitable, because currently, users can click the article's title or double-click the item to open the browser. 

There are already two entry points provided to the users, so there's no need to add another one.

Therefore, I will close this PR. If anyone has different opinions, feel free to comment :)

**Pull request closed without merging**